base/multimedia.jl

```diff
@@ -311,39 +351,49 @@ variants, one can also supply the "raw" data in the requested MIME type by passi
 application/postscript) or `x::Vector{UInt8}` (for binary MIME types).
 """
 function display(@nospecialize x)
-    for i = length(displays):-1:1
-        if xdisplayable(displays[i], x)
-            try
-                return display(displays[i], x)
-            catch e
-                isa(e, MethodError) && e.f in (display, show) ||
-                    rethrow()
+    for priority in sort(collect(keys(display_stack)), rev=true)
+        _displays = display_stack[priority]
+        for i = length(_displays):-1:1
+            if xdisplayable(_displays[i], x)
+                try
+                    return display(_displays[i], x)
+                catch e
+                    isa(e, MethodError) && e.f in (display, show) ||
+                        rethrow()
+                end
             end
         end
     end
     throw(MethodError(display, (x,)))
 end
 
 function display(m::MIME, @nospecialize x)
-    for i = length(displays):-1:1
-        if xdisplayable(displays[i], m, x)
-            try
-                return display(displays[i], m, x)
-            catch e
-                isa(e, MethodError) && e.f == display ||
-                    rethrow()
+    for priority in sort(collect(keys(display_stack)), rev=true)
+        _displays = display_stack[priority]
+        for i = length(_displays):-1:1
+            if xdisplayable(_displays[i], m, x)
+                try
+                    return display(_displays[i], m, x)
+                catch e
+                    isa(e, MethodError) && e.f == display ||
+                        rethrow()
+                end
             end
         end
     end
+
     throw(MethodError(display, (m, x)))
 end
 
 displayable(d::D, ::MIME{mime}) where {D<:AbstractDisplay,mime} =
     hasmethod(display, Tuple{D,MIME{mime},Any})
 
 function displayable(m::MIME)
-    for d in displays
-        displayable(d, m) && return true
+    for priority in sort(collect(keys(display_stack)), rev=true)
```

Comment:
It seems to me the order doesn't matter here? All you need to do is see if some display can handle it, right?

Reply:
True, I think I just copy-pasted that loop from `display` ;)